coderd/notifications/notifications_test.go

```diff
@@ -938,53 +985,263 @@ func TestNotificationTemplates_Golden(t *testing.T) {
+					if err != nil {
+						t.Logf("smtp not yet dialable: %s", err)
+						return false
+					}
 
-			wantBody, err := os.ReadFile(bodyGoldenFile)
-			require.NoError(t, err, fmt.Sprintf("missing golden notification body file. %s", hint))
-			wantTitle, err := os.ReadFile(titleGoldenFile)
-			require.NoError(t, err, fmt.Sprintf("missing golden notification title file. %s", hint))
+					if err = cl.Noop(); err != nil {
+						t.Logf("smtp not yet noopable: %s", err)
+						return false
+					}
+
+					if err = cl.Close(); err != nil {
+						t.Logf("smtp didn't close properly: %s", err)
+						return false
+					}
 
-			require.Equal(t, string(wantBody), body, fmt.Sprintf("rendered template body does not match golden file. If this is expected, %s", hint))
-			require.Equal(t, string(wantTitle), title, fmt.Sprintf("rendered template title does not match golden file. If this is expected, %s", hint))
+					return true
+				}, testutil.WaitShort, testutil.IntervalFast)
+
+				smtpCfg := defaultNotificationsConfig(database.NotificationMethodSmtp)
+				smtpCfg.SMTP = smtpConfig
+
+				smtpManager, err := notifications.NewManager(
+					smtpCfg,
+					*db,
+					defaultHelpers(),
+					createMetrics(),
+					logger.Named("manager"),
+				)
+				require.NoError(t, err)
+
+				smtpManager.Run(ctx)
+				time.Sleep(time.Second)
+
+				notificationCfg := defaultNotificationsConfig(database.NotificationMethodSmtp)
+
+				smtpEnqueuer, err := notifications.NewStoreEnqueuer(
+					notificationCfg,
+					*db,
+					defaultHelpers(),
+					logger.Named("enqueuer"),
+					quartz.NewReal(),
+				)
+				require.NoError(t, err)
+
+				_, err = smtpEnqueuer.EnqueueWithData(
+					ctx,
+					user.ID,
+					tc.id,
+					tc.payload.Labels,
+					tc.payload.Data,
+					user.Username,
+					user.ID,
+				)
+				require.NoError(t, err)
+
+				// Wait for the message to be fetched
+				var msg *mocksmtp.Message
+				require.Eventually(t, func() bool {
+					msg = backend.LastMessage()
+					return msg != nil
+				}, testutil.WaitShort, testutil.IntervalFast)
+				require.NotNil(t, msg, "want a message to be sent")
+
+				body := normalizeGoldenEmail([]byte(msg.Contents))
+
+				err = smtpManager.Stop(ctx)
+				require.NoError(t, err)
+
+				partialName := strings.Split(t.Name(), "/")[1]
+				goldenFile := filepath.Join("testdata", "rendered-templates", "smtp", partialName+".html.golden")
+				if *updateGoldenFiles {
+					err = os.MkdirAll(filepath.Dir(goldenFile), 0o755)
+					require.NoError(t, err, "want no error creating golden file directory")
+					err = os.WriteFile(goldenFile, body, 0o600)
+					require.NoError(t, err, "want no error writing body golden file")
+					return
+				}
+
+				wantBody, err := os.ReadFile(goldenFile)
+				require.NoError(t, err, fmt.Sprintf("missing golden notification body file. %s", hint))
+				require.Equal(t, string(wantBody), string(body), fmt.Sprintf("smtp notification does not match golden file. If this is expected, %s", hint))
```

Comment:
Suggestion: In my experience require has pretty hard to parse diffs, `cmp.Diff` can produce a bit nicer output (it's used in a few other places too).

Reply:
done